lib/scrolling_page_indicator.dart

```diff
@@ -184,24 +183,27 @@ class _Painter extends CustomPainter {
           }
 
           if (dot - _visibleFramePosition < currentScaleDistance) {
-            double calculatedDiameter = diameter * (dot - _visibleFramePosition) /
-                currentScaleDistance;
-            diameter = min(diameter, calculatedDiameter);
-          } else if (dot - _visibleFramePosition > width - currentScaleDistance) {
             double calculatedDiameter =
-                diameter * (-dot + _visibleFramePosition + width) / currentScaleDistance;
+                diameter * (dot - _visibleFramePosition) / currentScaleDistance;
+            diameter = min(diameter, calculatedDiameter);
+          } else if (dot - _visibleFramePosition >
```

Comment:
Bad formatting. Make line hard wrap at least 100-120 columns in your ide and then do reformat code.

Reply:
You can check it again, I've reformatted